### PR TITLE
Graph node class

### PR DESCRIPTION
**Pull request by raajtilaksarma**
Referred from PR #40 

#### Brief description of what is fixed or changed

Added graph class.
Added adjacency list inside `GraphNode` class as type `DynamicOneDimensionalArray`. The only problem is there is a function `DynamicOneDimensionalArray.delete()` which deletes by `id`. But what we need here is a function which deletes an element from a `DynamicOneDimensionalArray` using a value.
Currently a function `.remove()` is used.

### CODE REVIEW

**Comment by raajtilaksarma** [comment]
@czgdp1807  Can you make a function `remove()` in `DynamicOneDimensionalArray` which removes an element by value?

**Comment by raajtilaksarma** [comment]
I don't know why all checks have failed while creating this PR. Can you tell me why @czgdp1807 ?

**Comment by czgdp1807** [comment]
The problem in the removal of element by value is the increased amortized cost. Some research is needed to find the most optimized way to do that.

**Comment by raajtilaksarma** [comment]
Got it. I think we should have the type of adjacency list as a `set` then?
Cause if you add an edge your adjacency list for both nodes will change.
You need to update or remove your data structure by value. That value is
the key of the node. So we may not use dynamic one dimensional array?

Also, i will raise an issue for multigraphs

File: pydatastructs/utils/misc_util.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 from __future__ import print_function, division
+from pydatastructs import DynamicOneDimensionalArray as DODA
```
[the remark this answers is not in the document]
Comment (raajtilaksarma) [reply pydatastructs/utils/misc_util.py]: corrected it. Can you check it now? Also lmk if we are using `set`. I think i will be fast to add and remove elements.

**Comment by czgdp1807** [comment]
> I think we should have the type of adjacency list as a `set` then?

It needs some analysis. I will let you know my views on this by tonight.

**Comment by czgdp1807** [comment]
@raajtilaksarma I think we can go with `set` for now. 
```python
import sys, random, time
from pydatastructs.linear_data_structures import DynamicOneDimensionalArray
n = int(sys.argv[1])
max_val = 10000

elements_uniq = []
for _ in range(n):
    rn = random.randint(0, max_val)
    while rn in elements_uniq:
        rn = random.randint(0, max_val)
    elements_uniq.append(rn)

"""
set - remove and add
"""
start = time.process_time()
Set = set()
for element in elements_uniq:
    Set.add(element)
end = time.process_time()
print("Time taken to add %d elements is %f ns"%(n, (end - start)*10**9))
start = time.process_time()
for element in elements_uniq:
    Set.remove(element)
end = time.process_time()
print("Time taken to remove %d elements is %f ns"%(n, (end - start)*10**9))

"""
DynamicOneDimensionalArray with dict - append and delete
"""
start = time.process_time()
Array = DynamicOneDimensionalArray(int, 0)
indices = dict()
for element in elements_uniq:
    Array.append(element)
    indices[element] = Array._last_pos_filled
end = time.process_time()
print("Time taken to add %d elements is %f ns"%(n, (end - start)*10**9))
start = time.process_time()
for element in elements_uniq:
    Array.delete(indices[element])
    del indices[element]
end = time.process_time()
print("Time taken to remove %d elements is %f ns"%(n, (end - start)*10**9))
```

**Comment by czgdp1807** [comment]
ping @raajtilaksarma

**Comment by czgdp1807** [comment]
ping @raajtilaksarma Any news? Are you working on it?

**Comment by czgdp1807** [comment]
In accordance with the PR policy, this PR will be closed tomorrow. Thanks for the contributions.

**Comment by raajtilaksarma** [comment]
Had no internet access. Government removed it.
Please close the PR. Might not be able to make any contributions soon.

**Comment by czgdp1807** [comment]
Closing.

**Pull request closed without merging**